app/controllers/declaration/AdditionalFiscalReferencesController.scala

```diff
@@ -113,8 +112,10 @@ class AdditionalFiscalReferencesController @Inject()(
     cachedData: AdditionalFiscalReferencesData
   )(implicit request: JourneyRequest[_]): Future[Result] = {
     val updatedCache = MultipleItemsHelper.remove(values.headOption, cachedData.references)
-    updateExportsCache(itemId, journeySessionId, AdditionalFiscalReferencesData(updatedCache))
-      .map(_ => Ok(additionalFiscalReferencesPage(itemId, form.discardingErrors)))
+    updateExportsCache(itemId, AdditionalFiscalReferencesData(updatedCache)).map {
+      case Some(model) => Ok(additionalFiscalReferencesPage(itemId, form.discardingErrors))
+      case None        => Redirect(routes.ItemsSummaryController.displayPage())
```

Comment:
In what circumstance would the result of the call to updateExportsCache be Future(None)?

Reply:
Where there is delayed post after expiration for cache.

Reply:
If the item doesn't exist I think

Reply:
I don't see other controllers worrying about that.  Why would the item not exist?  It existed milliseconds earlier when the request was first handled.